spec/models/lesson_spec.rb

```diff
@@ -4,19 +4,21 @@
 
 RSpec.describe Lesson do
   before do
-    stub_user_info_api_for_teacher(teacher_id: User::TEACHER_ID, school_id: School::ID)
+    stub_user_info_api_for_teacher(teacher_id:, school_id: School::ID)
   end
 
+  let(:teacher_id) { SecureRandom.uuid }
```

Comment:
Minor: In a few places like this you've added calls to `let` *after* calls to `before`. I know there's nothing technically wrong with this, but I don't think it's as idiomatic/readable as having them the other way around.